File: emg-fe/src/useCases/CreateUserUseCase.js
import CreateUserConnector from '../connectors/CreateUserConnector.js';
import CreateUserUseCaseInterface from '../interfaces/useCases/CreateUserUseCaseInterface.js';

export default class CreateUserUseCase extends CreateUserUseCaseInterface {

    constructor(username, email, password) {
        super();
        this.username = username;
        this.email = email;
        this.password = password
        this.createUserConnector = new CreateUserConnector(this.username, this.email, this.password);
    }

    run() {
        return this.createUserConnector.createUser();
            
    }

    getUserId() {
        return this.createUserConnector.getUserId();
    }
}

File: waves/use_cases/create_user_use_case.py
from rest_framework.response import Response
from waves.interfaces.use_cases.create_user_use_case_interface import CreateUserUseCaseInterface
from waves.repositories.create_user_data_access import CreateUserDataAccess


class CreateUserUseCase(CreateUserUseCaseInterface):
    def __init__(self, username, email, password):
        self._username = username
        self._email = email
        self._password = password

    def run(self):
        create_user_data_access = CreateUserDataAccess(self._username, self._email, self._password)
        user_id = create_user_data_access.create_user()
        return Response(data=user_id, status=Response.status_code)
